components/cache/cache_invalidation.rst

```diff
@@ -0,0 +1,121 @@
+.. index::
+   single: Cache; Invalidation
+   single: Cache; Tags
+   single: Cache; Namespaces
+
+Cache Invalidation
+==================
+
+Cache invalidation is the process of removing all cached items related to a
+change in the state of your model. The most basic kind of invalidation is direct
+items deletion. But when the state of a primary resource has spread accross
+several cached items, keeping them in sync can be difficult.
+
+The Symfony Cache component provides three mechanisms to help solve this problem:
+
+* Tags based invalidation for managing data dependencies;
+* Namespace based invalidation for context dependend data;
+* Expiration based invalidation for time related dependencies.
+
+.. versionadded:: 3.2
+    Tags and namespace based invalidation was introduced in Symfony 3.2.
+
+Using Cache Tags
+----------------
+
+To benefit from the tags based invalidation, your responsiblity is to attach the
+proper tags to each cached items. Each tag is a plain string identifier that you
+can use to trigger the removal of all items that had this tag attached to them.
+
+To attach tags to cached items, you need to use the
+:method:`Symfony\\Component\\Cache\\CacheItem::tag` method that is implemented by
+cache items, as returned by cache adapters::
+
+    $item = $cache->getItem('cache_key');
+    // ...
+    // add one or more tags
+    $item->tag('tag_1');
+    $item->tag(array('tag_2', 'tag_3'));
+    $cache->save($item);
+
+If `$cache` implements :class:`Symfony\\Component\\Cache\\TagAwareAdapterInterface`,
+you can invalidate the cached items by calling
+:method:`Symfony\\Component\\Cache\\TagAwareAdapterInterface::invalidateTags`::
+
+    // invalidate all items related to `tag_1`
+    $cache->invalidateTags('tag_2');
+
+    // or invalidate all items related to `tag_1` or `tag_3`
+    $cache->invalidateTags(array('tag_1', 'tag_3'));
+
+Of course when you know the cache key, it's better to call
```

Comment:
I would reword that this way:

Using Cache tags are very useful when you don't need to know the cache key to delete.

And I will mention ``$cache->deleteItem('cache_key');`` in the previous example (self-explained).

WDYT ?